server/serializer.go

```diff
@@ -0,0 +1,219 @@
+package main
+
+import (
+	"encoding/json"
+	"io"
+
+	"github.com/pkg/errors"
+)
+
+type TelemetryAPIRequest struct {
+	Event      string
+	Properties map[string]interface{}
+}
+
+func GetTelemetryPayloadFromJSON(data io.Reader) (*TelemetryAPIRequest, error) {
+	body := &TelemetryAPIRequest{}
+	if err := json.NewDecoder(data).Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func IsTelemetryPayloadValid(t *TelemetryAPIRequest) error {
+	if t == nil {
+		return errors.New("invalid request body")
+	}
+
+	if t.Event == "" {
+		return errors.New("event is required")
+	}
+
+	return nil
+}
+
+type AddAPIRequest struct {
+	Message     string `json:"message"`
+	Description string `json:"description"`
+	SendTo      string `json:"send_to"`
+	PostID      string `json:"post_id"`
+}
+
+func GetAddIssuePayloadFromJSON(data io.Reader) (*AddAPIRequest, error) {
+	body := &AddAPIRequest{}
+	if err := json.NewDecoder(data).Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
+
+func IsAddIssuePayloadValid(a *AddAPIRequest) error {
+	if a == nil {
+		return errors.New("invalid request body")
+	}
+
+	if a.Message == "" {
+		return errors.New("message is required")
+	}
+
+	return nil
+}
+
+type EditAPIRequest struct {
+	ID          string `json:"id"`
+	Message     string `json:"message"`
+	Description string `json:"description"`
+}
+
+func GetEditIssuePayloadFromJSON(data io.Reader) (*EditAPIRequest, error) {
+	body := &EditAPIRequest{}
+	if err := json.NewDecoder(data).Decode(&body); err != nil {
+		return nil, err
+	}
+
+	return body, nil
+}
```

Comment:
0/5 Do we really need another method just for decoding?

Reply:
I have taken this reference from multiple plugins. One example is below
 https://github.com/mattermost/mattermost-plugin-servicenow/blob/bb7356be7639b8d8fb02f196ac753728d510d9cc/server/serializer/servicenow_incident.go#L40

Should i decode it inside the API function only?